flashinfer/quantization/mxfp8_layout.py

```
@@ -0,0 +1,306 @@
+def _ceil_div(x: int, y: int) -> int:
+    return (x + y - 1) // y
+
+
+def _ceil_to_ue8m0(x: torch.Tensor) -> torch.Tensor:
+    """UE8M0 ceil: round x to the next power of two."""
+    return torch.pow(2.0, torch.ceil(torch.log2(x.abs())))
+
+
+@flashinfer_api
+def mxfp8_quantize_per_token(
+    input: torch.Tensor,
```

Ruff (0.15.15)

[error] Function argument `input` is shadowing a Python builtin

(A002)